### PR TITLE
Update SFTP info

### LINKED ISSUE #3336: Document SFTP availability on Grid
### Where on docs.platform.sh should be changed?

TBD

### What exactly should be updated?

We should make it known that SFTP is available on Grid via the built-in SSH.
User has access to all files. See context.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AnouckColson**
## Why

Closes #3336 

## What's changed

Updated DG, DG3 sections and File transfer page to document SFTP behaviour and limitations on each architecture.

### CODE REVIEW

File: sites/platform/src/development/file-transfer.md [hunk 1]
```diff
@@ -166,3 +174,70 @@ If you're using UTF-8 encoded files on macOS,
...
+The `sftp` connection is open once the `sftp>` prompt is displayed in your terminal.
+
+#### Download a file
+
+Say you want to download a `diagram.png` file from the `web/uploads` directory 
+(relative to the [app root](../create-apps/app-reference.md#root-directory)).
+To do so, run the following command:
+
+```
+sftp> get /app/web/uploads/diagram.png
```
Comment (gilzow) [inline sites/platform/src/development/file-transfer.md:228]: you dont _have_ to use the whole path to the file for the `get` and `put` commands. you can use relative paths, or `cd` into the target directory before issuing the command. But the full path should always work.

File: sites/platform/src/development/file-transfer.md [hunk 2]
```diff
@@ -166,3 +174,70 @@ If you're using UTF-8 encoded files on macOS,
...
+The `diagram.png` file is copied to the current local directory.
+
+#### Upload a file
+
+Say you want to upload a `diagram.png` file to the `web/uploads` directory 
+(relative to the [app root](../create-apps/app-reference.md#root-directory)).
+To do so, run the following command:
+
+```bash
+sftp> put diagram.png to /app/web/uploads
```
Comment (gilzow) [inline sites/platform/src/development/file-transfer.md:240]: same as previous that the user isn't required to use the full path.

File: sites/platform/src/dedicated-gen-2/architecture/options.md [hunk 3]
```diff
@@ -30,12 +33,13 @@ The cluster begins to look more like a standard N-Tier architecture at this poin
...
 This configuration requires a separate setup from the default so advanced planning is required.
 
-## SFTP accounts
+## `sftp`
 
-In addition to SSH accounts, SFTP accounts can be created with a custom user/password that are restricted to certain directories. 
-These directories must be one of the writeable mounts (or rather, there’s no point assigning them to the read-only code directory). 
-There is no cost for this configuration, and it can be requested at any time via a support ticket. 
-SSH public key based authentication is also supported on the SFTP account.
+In addition to SSH accounts, `sftp` accounts can be created with a custom user/password that are restricted to certain directories.
```
Comment (gilzow) [inline sites/platform/src/dedicated-gen-2/architecture/options.md:38]: really? TIL. and that would be incredibly useful on the grid!

Comment (gilzow) [reply sites/platform/src/dedicated-gen-2/architecture/options.md:38]: Should we link from this section back over to /development/file-transfer.html#sftp ?

File: sites/platform/src/dedicated-gen-3/options.md [hunk 4]
```diff
@@ -23,4 +33,17 @@ But note that multiple-AZ configurations don't improve the contractual 99.99% up
 {{% vendor/name %}} is responsible for meeting the 99.99% uptime SLA no matter what,
 so multiple-AZ deployments should only be considered in cases where they're truly appropriate.
 
-Multi-AZ deployments are available only on select AWS regions.
\ No newline at end of file
+Multi-AZ deployments are available only on select AWS regions.
+
+## `sftp`
+
+In addition to SSH accounts, you can create `sftp` accounts with a custom user/password.
```
Comment (gilzow) [inline sites/platform/src/dedicated-gen-3/options.md:40]: similar to the other comment: should we link from this section over to /development/file-transfer.html#sftp?

**Review by gilzow — DISMISSED** [review]
Approved with consideration given to the comments that have been left

File: sites/platform/src/development/file-transfer.md (hunk 1 above)
Comment (AnouckColson) [reply sites/platform/src/development/file-transfer.md]: Yes, now that you pointed this out, I think it makes more sense to use relative paths in the examples so users for ease of use/so users know they aren't required to use the whole path + consistency with `rsync` and `scp` examples.

File: sites/platform/src/dedicated-gen-2/architecture/options.md (hunk 3 above)
Comment (AnouckColson) [reply sites/platform/src/dedicated-gen-2/architecture/options.md]: Absolutely!

**Review by gilzow — DISMISSED** [review]

**Review by Mupsi — APPROVED** [review]

**Pull request merged**